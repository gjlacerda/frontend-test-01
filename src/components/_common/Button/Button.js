import React from 'react'
import styled from 'styled-components'
import { colorApp, colorAppDark } from 'utils/color'

const Button = styled.button`
  border: 1px solid ${colorApp};
  background: none;
  color: ${colorApp};
  padding: 10px 30px;
  width: ${props => props.fullWidth ? '100%' : 'auto'};
  text-transform: uppercase;
  cursor: pointer;
  transition: color 0.1s ease-in-out;
  transition-property: color, background-color;

  &:hover {
    background-color: ${colorApp};
    color: #fff;
  }

  &:active {
    background-color: ${colorAppDark};
  }
`

export default ({
  fullWidth,
  type,
  onClick,
  disabled,
  children,
}) => (
  <Button
    fullWidth={fullWidth}
    type={type}
    disabled={disabled}
    onClick={onClick}
  >
    {children}
  </Button>
)
