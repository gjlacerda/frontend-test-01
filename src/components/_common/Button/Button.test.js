import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import Button from './Button'

describe('Button component', () => {
  it('deve ser igual ao snapshot', () => {
    const tree = renderer.create(
      <Button />,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('deve ser igual ao snapshot quando passado o parÃ¢metro (fullWidth)', () => {
    const tree = renderer.create(
      <Button fullWidth />,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
