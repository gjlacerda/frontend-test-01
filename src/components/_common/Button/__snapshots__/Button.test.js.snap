// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component deve ser igual ao snapshot 1`] = `
.c0 {
  border: 1px solid #A99E7E;
  background: none;
  color: #A99E7E;
  padding: 10px 30px;
  width: auto;
  text-transform: uppercase;
  cursor: pointer;
  -webkit-transition: color 0.1s ease-in-out;
  transition: color 0.1s ease-in-out;
  -webkit-transition-property: color,background-color;
  transition-property: color,background-color;
}

.c0:hover {
  background-color: #A99E7E;
  color: #fff;
}

.c0:active {
  background-color: #766f58;
}

<button
  className="c0"
/>
`;

exports[`Button component deve ser igual ao snapshot quando passado o par√¢metro (fullWidth) 1`] = `
.c0 {
  border: 1px solid #A99E7E;
  background: none;
  color: #A99E7E;
  padding: 10px 30px;
  width: 100%;
  text-transform: uppercase;
  cursor: pointer;
  -webkit-transition: color 0.1s ease-in-out;
  transition: color 0.1s ease-in-out;
  -webkit-transition-property: color,background-color;
  transition-property: color,background-color;
}

.c0:hover {
  background-color: #A99E7E;
  color: #fff;
}

.c0:active {
  background-color: #766f58;
}

<button
  className="c0"
/>
`;
